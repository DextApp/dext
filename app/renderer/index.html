<!doctype html>
<html>
  <head>
    <title>Dext</title>
    <style type="text/css">
      body,
      html {
        margin: 0;
        padding: 0;
        background-color: transparent;
      }
      #app {
        background-color: transparent;
      }
    </style>
  </head>

  <body>

    <div id="app"></div>

    <script type="text/javascript" src="lib/bundle.js"></script>

  </body>
</html>
