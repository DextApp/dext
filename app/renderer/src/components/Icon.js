import React, { PropTypes } from 'react';
import { compose, style } from 'glamor';

const iconImg = style({
  maxWidth: 40,
  maxHeight: 40,
});

const iconImgText = compose(
  iconImg,
  style({
    display: 'block',
    backgroundColor: '#888',
    color: '#fff',
    textDecoration: 'none',
    textAlign: 'center',
    width: 40,
    height: 40,
    lineHeight: '40px',
    fontSize: 20,
    fontWeight: 'bold',
    borderRadius: 20,
  }),
);

const Icon = (props) => {
  switch (props.icon.type) {
    case 'text':
      return <span {...iconImgText} role="presentation">{props.icon.letter}</span>;
    case 'file':
      // no break
    default: // eslint-disable-line no-fallthrough
      return <img {...iconImg} src={props.icon.path} role="presentation" />;
  }
};

Icon.defaultProps = {
  icon: {
    type: '',
    path: '',
    letter: '',
  },
};

Icon.propTypes = {
  icon: PropTypes.shape({
    type: PropTypes.oneOf(['file', 'text', '']),
    path: PropTypes.string,
    letter: PropTypes.string,
  }),
};

export default Icon;
