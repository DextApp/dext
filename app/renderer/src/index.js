/* global document */
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import DevTools from './components/DevTools';
import AppContainer from './containers/AppContainer';
import configureStore from './store';

const store = configureStore();

// load the dev tools if necessary
let devTools = '';
if (process.env.NODE_ENV === 'development') {
  devTools = <DevTools />;
}

const DextApp = () => (
  <Provider store={store}>
    <div>
      <AppContainer />
      {devTools}
    </div>
  </Provider>
);

const rootNode = document.getElementById('app');

ReactDOM.render(<DextApp />, rootNode);
