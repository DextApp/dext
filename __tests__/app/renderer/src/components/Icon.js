import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../../../../../app/renderer/src/components/Icon';

describe('<Icon /> component', () => {
  it('should render a text icon component', () => {
    const icon = {
      type: 'text',
      letter: 'T',
    };
    const component = renderer.create(
      <Icon icon={icon} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a file icon component', () => {
    const icon = {
      type: 'file',
      letter: './icon.png',
    };
    const component = renderer.create(
      <Icon icon={icon} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
