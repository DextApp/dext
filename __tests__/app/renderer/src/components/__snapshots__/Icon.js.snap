exports[`<Icon /> component should render a file icon component 1`] = `
<img
  data-css-1pj9gkn=""
  role="presentation"
  src={undefined} />
`;

exports[`<Icon /> component should render a text icon component 1`] = `
<span
  data-css-1n4qoom=""
  role="presentation">
  T
</span>
`;
